site/layouts/shortcodes/codeFromFile.html

```diff
@@ -2,25 +2,41 @@
 {{ $lang := "" }}
 {{ $suffix := findRE "(\\.[^.]+)$" $file 1 }}
 {{ with  $suffix }}
-{{ $lang = (index . 0 | strings.TrimPrefix ".") }}
+  {{ $lang = (index . 0 | strings.TrimPrefix ".") }}
 {{ end }}
 {{ with .Get "lang" }}{{ $lang = . }}{{ end }}
-{{ $virtualPath :=  strings.TrimPrefix `static/` $file }}
+{{ $virtualPath := strings.TrimPrefix `static/` $file }}
 <table class="includecode" id="code-{{ $virtualPath }}">
   <thead>
     <tr>
       <th>
         <a href="/{{ $virtualPath }}">{{ $virtualPath }}</a>
-        <button onclick='copyText("code-{{ $virtualPath }}-hidden-copy-text");' title="Copy to clipboard">
+        <button onclick='copyText("code-{{ $virtualPath }}-hidden-copy-text", "button-{{ $virtualPath }}");' title="Copy to clipboard">
           <img src="{{ "copycode.svg" | relURL }}" alt="Copy">
         </button>
+        <span class="copy-success" id="button-{{ $virtualPath }}-success"></span>
       </th>
     </tr>
   </thead>
   <tbody>
     <tr>
-      <td>{{- highlight (trim ($file | readFile) "\n") $lang "" | -}}</td>
+      <td>{{- highlight (trim ($file | readFile) "\n") $lang "" -}}</td>
     </tr>
   </tbody>
-  <textarea class="hidden-copy-text" id="code-{{ $virtualPath }}-hidden-copy-text">{{ ($file | readFile) }}</textarea>
+  <textarea class="hidden-copy-text" id="code-{{ $virtualPath }}-hidden-copy-text">{{ $file | readFile }}</textarea>
 </table>
+
+<script>
+  function copyText(elementId, successElementId) {
+    const textarea = document.getElementById(elementId);
+    textarea.select();
```

Comment:
`textarea` is possibly `undefined` if the HTML has been modified by either extensions or generated incorrectly. Should we at least determine it is safe to be accessed before using it?

```suggestion
    const textarea = document.getElementById(elementId);
    if (!textarea) {
      // perhaps some console.warn handling?
      return
    }
    textarea.select();
```